nodeup/pkg/model/docker.go

```diff
@@ -43,270 +43,9 @@ var _ fi.ModelBuilder = &DockerBuilder{}
 
 // DefaultDockerVersion is the (legacy) docker version we use if one is not specified in the manifest.
 // We don't change this with each version of kops, we expect newer versions of kops to populate the field.
-const DefaultDockerVersion = "1.12.3"
+const DefaultDockerVersion = "17.03.2"
```

Comment:
Might be better to instead require a value in the manifest.

Reply:
This is used as a fallback when the dirstro or arch is not listed. For example, if anyone chooses to use a Debian Jessie based image with latest Docker, NodeUp will not find it and fall back to this.

Reply:
How does that work? I only see this referenced  from the `dockerVersion()` receiver of `DockerBuilder`, when `Cluster.Spec.Docker.Version` is not specified.

It seems odd to be updating a field to the oldest version we can. The comment suggests it's a backwards-compatibility strategy that is never supposed to be updated.

Reply:
I looked at the code and I also don't see any other reference. It is used just if the version doesn't exist as you also noticed.

This is a very old piece of code that exists since the beginnings. No idea why it was needed then and why we still keep it. At this point there should always be a Docker version.
https://github.com/kubernetes/kops/blob/62c51002ba69aaafeff38191ad02a8a7b6a2ba1d/pkg/model/components/docker.go#L50-L62

The only explanation I have is this, since containerd support was added: https://github.com/kubernetes/kops/pull/7986#discussion_r357925590.

Reply:
Yeah, the `components/docker.go` code that presumably ensures the field is set was added in the same commit that adds `DefaultDockerVersion`. @justinsb would have the context.

Reply:
This is indeed very very old.  The kops CLI is supposed to populate the docker version in the "expanded" form of the cluster spec (the one you can see with `--full`).  When it was first introduced, we didn't lock nodeup to the kops version, and so there was the risk of skew.  To tolerate that, we introduced a default docker version that nodeup would use, when kops didn't populate it.

At this point, it's probably easier to remove the nodeup defaulting entirely - just skip installing docker if isn't populated, I would suggest (with a warning so we can figure out what happened!).